testsrc/org/mozilla/javascript/tests/es6/NativeRegExpTest.java

```diff
@@ -148,16 +131,41 @@ public void regExIsNotCallable() {
 
     @Test
     public void lastIndexReadonly() {
-        final String script = "try { "
-                + "  var r = /c/g;"
-                + "  Object.defineProperty(r, 'lastIndex', { writable: false });"
-                + "  r.exec('abc');"
-                + "} catch (e) { e.message }";
-        Utils.runWithAllOptimizationLevels(_cx -> {
-            final ScriptableObject scope = _cx.initStandardObjects();
-            final Object result = _cx.evaluateString(scope, script, "test script", 0, null);
-            assertEquals("Cannot modify readonly property: lastIndex.", Context.toString(result));
-            return null;
-        });
+        final String script =
+                "try { "
+                        + "  var r = /c/g;"
+                        + "  Object.defineProperty(r, 'lastIndex', { writable: false });"
+                        + "  r.exec('abc');"
+                        + "} catch (e) { e.message }";
+        Utils.runWithAllOptimizationLevels(
+                _cx -> {
+                    final ScriptableObject scope = _cx.initStandardObjects();
+                    final Object result = _cx.evaluateString(scope, script, "test script", 0, null);
+                    assertEquals(
+                            "Cannot modify readonly property: lastIndex.",
+                            Context.toString(result));
+                    return null;
+                });
+    }
+
+    @Test
+    public void search() {
+        Context cx = Context.enter();
```

Comment:
Can you please turn this into a try..catch with resources, like this:

try (Context cx = Context.enter()) { // Do the tests }

Otherwise if your tests fail or throw an exception then the context leaks and that can create problems for other tests -- we already have to run all our tests in separate JVMs because some tests leak contexts and it'd be really nice to be able to stop doing that.

Reply:
Will be great if you can do this improvement for all tests in this file.

Reply:
Do I still need to do `Context.exit()` when using try-with-resources here?

Reply:
Looks like I do not, great!